ios/FluentUI/Core/SwiftUI+ViewPresentation.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// Modifier that presents a SwiftUI View as an additional layer on top of the content View.
+struct PresentationModifier<PresentedView: View>: ViewModifier {
+    func body(content: Content) -> some View {
+        ZStack {
+            content
+
+            // Backdrop for the presented view which overlays the
+            // content where it's being presented from.
+            Rectangle()
+            .frame(maxWidth: .infinity,
+                   maxHeight: .infinity,
+                   alignment: .center)
+            .foregroundColor(isPresented ? backgroundColor : Color.clear)
+            .allowsHitTesting(isPresented && isBlocking)
```

Comment:
Do we want to allow interaction with dimmed background? (i.e. should we add `&& !hasDimmedBackground` here or leave that up to implementer to decide)

Reply:
I thought about that and decided to leave it very flexible so that the implementer would decide (as you mentioned).
I'm thinking of a scenario where the dimmedBackgroundColor would be very transparent to even make sense... 
Do you have any concerns or a scenario that this could lead to undesired/unintended consequences?

Reply:
No concerns! Now that I'm thinking about it, I actually think it may be good for future controls like Teaching Callout where tapping on the background can dismiss the control entirely like below:

<img width="720" alt="image" src="https://user-images.githubusercontent.com/31874971/160207337-e8f835c9-0a76-48ba-9fa8-c9aa37b24563.png">